frontend/src/pages/main/home/client/components/graphic/Graphic.tsx

```diff
@@ -0,0 +1,117 @@
+        setIncome(income);
+    }
+
+    const mergeTransactions = (transactions: Transaction[]): Transaction[] => {
+        const mergedTransactions: { [nameTo: string]: Transaction } = {};
+        const paymentTransactions = transactions.filter(transaction => transaction.TransactionType === TransactionType.PAYMENT);
+        let numberUniqueTransactions = 0;
+        let flag = true;
+
+        paymentTransactions.forEach(transaction => {
+            const key = transaction.nameTo;
+            if (!mergedTransactions[key] && numberUniqueTransactions < THRESHOLD_UNIQUE_TRANSACTIONS) {
+                mergedTransactions[key] = {...transaction};
+                numberUniqueTransactions++;
+            } else if (!mergedTransactions[key] && flag) {
+                mergedTransactions[OTHER_TRANSACTIONS] = {...transaction, nameTo: OTHER_TRANSACTIONS};
+                flag = false;
+            } else if (!mergedTransactions[key]) {
+                mergedTransactions[OTHER_TRANSACTIONS].amount += transaction.amount;
+            } else {
+                mergedTransactions[key].amount += transaction.amount;
+            }
+        });
+        return Object.values(mergedTransactions);
+    };
+
+    const graphicData = {
+        labels:
+            transactions.map(transaction => {
+                const percentage = Math.round((transaction.amount / expenses) * 100);
+                return `${transaction.nameTo}: ${transaction.amount}${CURRENCY_USD} (${percentage}%)`;
+            }),
+        datasets: [
+            {
+                data: transactions.map((transaction) => transaction.amount),
+                backgroundColor: COLORS,
+                borderColor: "#ffffff",
+                hoverOffSet: 4
+            }
+        ]
+    }
+
+    useEffect(() => {
+        updateTransactions().then();
+    }, []);
+
+    return (
+        <Card className="card-background card-payment card-graphic">
+            <CardContent className="card-content">
+                <Typography
+                    variant="h4"
+                >
+                    {Content.TRANSACTIONS}
+                </Typography>
+            </CardContent>
+            <Finance income={income} expenses={expenses}/>
+            {expenses > 0 ? <Pie options={{
+                    responsive: true,
+                    plugins: {
+                        legend: {
+                            display: true,
+                            position: 'left',
+                            align: "center",
+                            labels: {
+                                usePointStyle: true,
+                                font: {
+                                    size: 14,
+                                    weight: "bold"
+                                }
+                            }
+                        }
+                    }
+                }} data={graphicData} className="graphic"/>
```

Comment:
it goes without saying you have to extract configs and maybe the whole component.